### PR TITLE
Source Zendesk Support: refactor and debug stuck syncs

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/1231

## How
 - Reduce duplicate sleeps in a main thread
 - Reduce backoff time (for unknown reason it was doubled)
 - Added warning if a ProcessPoolExecutor is used instead of a ThreadPoolExecutor
 - Added more logs to understand what makes the sync stuck
 - Introduced `ignore_pagination` config variable to speed up SATs

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-zendesk-support/unit_tests/test_futures.py [hunk 1]
```diff
@@ -131,3 +132,45 @@ def record_gen(start=0, end=page_size):
...
+                "text": json.dumps({"organizations": list(record_gen(page * page_size, min(records_count, (page + 1) * page_size)))})
+            }
+            for page in range(pages)
+        ]
+        m.get(records_url, responses)
+        records = list(stream.read_records(sync_mode=SyncMode.full_refresh))
+        assert len(records) == records_count
+        end = datetime.datetime.now()
+        sleep_time = int(60 / x_rate_limit)
+        assert sleep_time - 1 <= (end - start).seconds <= sleep_time + 1
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-zendesk-support/unit_tests/test_futures.py:176]: this test shows 6 seconds sleeping against 44 on a previous version

**Review by bazarnov — APPROVED** [review]
Please move the common method from the stream class to their base class.

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**